### PR TITLE
jax.scipy.signal functions for n-dim arrays

### LINKED ISSUE #4775: jax.scipy.signal.convolve should support n-dimensional inputs
`scipy.signal.convolve()` supports N-dimensional inputs; the JAX equivalent should as well.

There is already a `_convolve_nd` utility function in the module; this would involve calling that, and writing a test case to ensure the outputs match scipy's.

### PR DESCRIPTION
**Pull request by ashridh**
Fixes: #4775 .

Changes include:

1. `jax.scipy.signal.correlate` was giving incorrect results for tensors with ndim >= 2. Fixed this.
2. Added `_convolve_nd` in definition of `convolve()` and `correlate()`
3. Added tests for running these functions with 2d arrays.
4. Fix a small error where `jnp.scipy.signal.correlate2d` was wrapping `scipy.signal.correlate`.

### CODE REVIEW

File: tests/scipy_signal_test.py [hunk 1]
```diff
@@ -49,10 +49,12 @@ class LaxBackedScipySignalTests(jtu.JaxTestCase):
       for mode in ['full', 'same', 'valid']
       for op in ['convolve', 'correlate']
       for dtype in default_dtypes
-      for xshape in onedim_shapes
-      for yshape in onedim_shapes))
+      for xshape in onedim_shapes + twodim_shapes
+      for yshape in onedim_shapes + twodim_shapes))
```
Comment (jakevdp) [inline tests/scipy_signal_test.py:53]: It would be worth also testing some three-dimensional shapes as well.

**Review by jakevdp — COMMENTED** [review]
Looks great, thanks for taking this on! Just a few more tests and I think this will be good to go.

File: tests/scipy_signal_test.py [hunk 2]
```diff
@@ -49,10 +50,12 @@ class LaxBackedScipySignalTests(jtu.JaxTestCase):
...
       for op in ['convolve', 'correlate']
       for dtype in default_dtypes
-      for xshape in onedim_shapes
-      for yshape in onedim_shapes))
+      for xshape in onedim_shapes + twodim_shapes + threedim_shapes
+      for yshape in onedim_shapes + twodim_shapes + threedim_shapes))
   def testConvolutions(self, xshape, yshape, dtype, mode, jsp_op, osp_op):
     rng = jtu.rand_default(self.rng())
+    if xshape != yshape:
+      return
```
Comment (jakevdp) [inline tests/scipy_signal_test.py:58]: This seems strange to only test cases where the two arrays have equal shape. Maybe instead in the parametrization we can do something like:
```python
for shapeset in [onedim_shapes, twodim_shapes, threedim_shapes]
for xshape in shapeset
for yshape in shapeset
```
Then tests will cover mixed pairs of shapes with the same dimension.

Comment (ayush-1506) [reply tests/scipy_signal_test.py]: Makes sense, will do!

**Review by jakevdp — APPROVED** [review]
Looks great, thanks!
